src/udev/udevadm-info.c

```diff
@@ -858,6 +920,70 @@ int info_main(int argc, char *argv[], void *userdata) {
                                 return r;
                         break;
 
+                case ARG_SUBSYSTEM_MATCH:
+                case ARG_SUBSYSTEM_NOMATCH:
+                        r = sd_device_enumerator_add_match_subsystem(e, optarg, c == ARG_SUBSYSTEM_MATCH);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to add%s subsystem match '%s': %m",
+                                                       c == ARG_SUBSYSTEM_MATCH ? "" : " negative", optarg);
+
+                        break;
+
+                case ARG_ATTR_MATCH:
+                case ARG_ATTR_NOMATCH:
+                        r = parse_key_value_argument(optarg, &k, &v);
```

Comment:
this leaks memory if used twice on the same cmdline. k/v should be declared in a local `{}` block here, so that they are freed right away, after we used them